src/index.js

```diff
@@ -747,6 +747,39 @@ walletRouter.post(
   }
 );
 
+/**
+ * GET request to obtain adress information
+ * For the docs, see api-docs.js
+ */
+ walletRouter.get(
+  '/address-info',
+  query('address').isString(),
+  query('token').isString().optional(),
+  (req, res) => {
+    try {
+      // Query parameters validation
+      const validationResult = parametersValidation(req);
+      if (!validationResult.success) {
+        return res.status(400).json(validationResult);
+      }
+
+      const wallet = req.wallet;
+      const { address, token } = matchedData(req, { locations: ['query'] });
+      const result = wallet.getAddressInfo(address, { token });
+      res.send({
+        success: true,
+        ...result
+      });
+    } catch(error) {
+      if (error instanceof errors.AddressError) {
+        res.send({ success: false, error: error.message });
+      }
+
+      throw error;
```

Comment:
Shouldn't it be in the `else` block? I mean, we shouldn't re-raise the exception if it was handled returning the error to the user, right?

Reply:
Absolutely right, thanks.